ydb/mvp/oidc_proxy/oidc_protected_page_n.h

```diff
@@ -0,0 +1,340 @@
+#pragma once
+
+#include <util/generic/hash_set.h>
+#include <ydb/library/actors/core/actorsystem.h>
+#include <ydb/library/actors/core/actor.h>
+#include <ydb/library/actors/core/hfunc.h>
+#include <ydb/library/actors/core/events.h>
+#include <ydb/library/actors/core/event_local.h>
+#include <ydb/library/actors/core/actor_bootstrapped.h>
+#include <ydb/library/actors/http/http.h>
+#include <ydb/public/lib/deprecated/client/grpc_client.h>
+#include <ydb/library/grpc/client/grpc_client_low.h>
+#include <ydb/library/actors/core/log.h>
+#include <ydb/public/api/client/yc_private/oauth/session_service.grpc.pb.h>
+#include <ydb/mvp/core/mvp_log.h>
+#include <ydb/mvp/core/mvp_tokens.h>
+#include <ydb/mvp/core/appdata.h>
+#include <ydb/mvp/core/protos/mvp.pb.h>
+#include <ydb/core/util/wildcard.h>
+#include "openid_connect.h"
+
+namespace NMVP {
+
+class THandlerSessionServiceCheckN : public NActors::TActorBootstrapped<THandlerSessionServiceCheckN> {
+private:
+    using TBase = NActors::TActorBootstrapped<THandlerSessionServiceCheckN>;
+    using TSessionService = yandex::cloud::priv::oauth::v1::SessionService;
+
+    const NActors::TActorId Sender;
+    const NHttp::THttpIncomingRequestPtr Request;
+    NActors::TActorId HttpProxyId;
+    const TOpenIdConnectSettings Settings;
+    TString ProtectedPageUrl;
+    TString RequestedPageScheme;
+    bool  IsAjaxRequest = false;
+
+    const static inline TStringBuf NOT_FOUND_HTML_PAGE = "<html><head><title>404 Not Found</title></head><body bgcolor=\"white\"><center><h1>404 Not Found</h1></center></body></html>";
```

Comment:
This is the same constants and similar code here. I think that we need to reuse it and split into base and derived classes